client/client.go

```diff
@@ -53,7 +53,11 @@ func New(address string, opts ...ClientOpt) (*Client, error) {
 	if address == "" {
 		address = appdefaults.Address
 	}
-	conn, err := grpc.Dial(address, gopts...)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
```

Comment:
I'm a bit late to the game here, but changing this 30s (which is quite long in some cases, the cli just sits there for 30s if there is no backend running) has been on my todo list, if it's not considered generically appropriate to shorten it perhaps it could be exposed as an option -- maybe the best way would be to take the `ctx` as an option to `New` and let the caller make it one with a timeout if they want?

Reply:
+1, could you open a PR?
Probably we want a CLI flag as well.

Reply:
Will open a PR. Any initial thoughts on adding a new timeout option vs exposing `ctx`? I'm leaning towards the latter (perhaps retaining the timeout here as a backstop)

Reply:
+1 for the latter

Reply:
https://github.com/orisano/buildkit/blob/timeout-option/client/client.go
like this?

Reply:
That would work too, but I think passing in a `ctx` will also have the advantage that any `Ctrl-C` handling which the client ass hooked onto its context will also cancel down this chain. Right now pressing Ctrl-C on `buildctl` can take a little while.

Reply:
I see.

Reply:
I implemented another version.
https://github.com/orisano/buildkit/commit/7c7642ae2692149a1e9584194707c2e1e045db0c

Reply:
I already opened #440 as promised in https://github.com/moby/buildkit/pull/432#discussion_r193357441.